docs/documentation/upgrading/topics/changes/changes-26_0_0.adoc

```diff
@@ -183,3 +183,10 @@ Additionally, the following resources have been removed from the `common` theme:
 - `node_modules/jquery`
 
 If you previously used any of the removed resources in your theme, make sure to add them to your own theme resources instead.
+
+= Additional datasources now require using XA
+
+{project_name} by default does not use XA transactions. However, this is considered unsafe if more than one datasource is used.
+Starting with this release, you need to use XA transactions if you are adding additional datasources to {project_name}.
+For the default datasource, you can do this by setting the `--transaction-xa-enabled=true` option. For additional datasources, you need to use
+the `quarkus.datasource.<your-datasource-name>.jdbc.transactions=xa` option in your `quarkus.properties` file.
```

Comment:
It should also mention that you'll get a warning about recovery not being enabled, and likely mention that there isn't out-of-the box support via the operator.

Reply:
I don't think I noticed any warning.

Operator supports the [`.spec.transaction. xaEnabled` field](https://github.com/keycloak/keycloak/blob/98dc21e6fc2b5a46b18d8ee5f9f1d271d8bf5ffe/operator/src/main/java/org/keycloak/operator/crds/v2alpha1/deployment/spec/TransactionsSpec.java).

Reply:
Added a note on the recovery.

Reply:
> Added a note on the recovery.

We should expand on that with a WARNING and call out containerized / operator usage specifically. If it's not clear when the full xa features are not available, then this solution is not much better than setting the workaround flag.

cc @yrodiere

Reply:
@shawkins I wonder if we could do that as a follow-up. This change here is to just document the breaking change we did that XA is now required. The upgrading guide is IMHO not supposed to supplement a proper docs for additional datasources which will be added as part of https://github.com/keycloak/keycloak/issues/29116. On a side note, AFAIK, Quarkus itself doesn't provide such warning in their docs.

Reply:
> I wonder if we could do that as a follow-up.

Sounds good.

> On a side note, AFAIK, Quarkus itself doesn't provide such warning in their docs.

I agree https://quarkus.io/guides/transaction does not spell out the containerized scenario, but it seems like it could be less obvious or assumed to be turn-key when utilized via Keycloak so we should be clear about expectations.

Reply:
Added a comment in https://github.com/keycloak/keycloak/issues/29116 to remind us later.